--- libtool.orig	2003-12-12 16:02:52.000000000 +0100
+++ libtool	2003-12-12 16:11:30.000000000 +0100
@@ -191,8 +191,8 @@
 
 # Commands used to build and install an old-style archive.
 RANLIB="ranlib"
-old_archive_cmds="\$AR \$AR_FLAGS \$oldlib\$oldobjs\$old_deplibs~\$RANLIB \$oldlib"
-old_postinstall_cmds="\$RANLIB \$oldlib~chmod 644 \$oldlib"
+old_archive_cmds="\$AR \$AR_FLAGS \$oldlib\$oldobjs\$old_deplibs!\$RANLIB \$oldlib"
+old_postinstall_cmds="\$RANLIB \$oldlib!chmod 644 \$oldlib"
 old_postuninstall_cmds=""
 
 # Create an old-style archive from a shared archive.
@@ -2618,7 +2618,7 @@
 	    if test -f "$output_objdir/$soname-def"; then :
 	    else
 	      $show "extracting exported symbol list from \`$soname'"
-	      save_ifs="$IFS"; IFS='~'
+	      save_ifs="$IFS"; IFS='!'
 	      eval cmds=\"$extract_expsyms_cmds\"
 	      for cmd in $cmds; do
 		IFS="$save_ifs"
@@ -2631,7 +2631,7 @@
 	    # Create $newlib
 	    if test -f "$output_objdir/$newlib"; then :; else
 	      $show "generating import library for \`$soname'"
-	      save_ifs="$IFS"; IFS='~'
+	      save_ifs="$IFS"; IFS='!'
 	      eval cmds=\"$old_archive_from_expsyms_cmds\"
 	      for cmd in $cmds; do
 		IFS="$save_ifs"
@@ -3888,7 +3888,7 @@
 	    export_symbols="$output_objdir/$libname.exp"
 	    $run $rm $export_symbols
 	    eval cmds=\"$export_symbols_cmds\"
-	    save_ifs="$IFS"; IFS='~'
+	    save_ifs="$IFS"; IFS='!'
 	    for cmd in $cmds; do
 	      IFS="$save_ifs"
 	      if len=`expr "X$cmd" : ".*"` &&
@@ -4065,7 +4065,7 @@
 	      else
 		# All subsequent reloadable object files will link in
 		# the last one created.
-		eval concat_cmds=\"\$concat_cmds~$reload_cmds $objlist $last_robj\"
+		eval concat_cmds=\"\$concat_cmds!$reload_cmds $objlist $last_robj\"
 	      fi
 	      last_robj=$output_objdir/$save_output-${k}.$objext
 	      k=`expr $k + 1`
@@ -4077,7 +4077,7 @@
 	  # Handle the remaining objects by creating one last
 	  # reloadable object file.  All subsequent reloadable object
 	  # files will link in the last one created.
-	  test -z "$concat_cmds" || concat_cmds=$concat_cmds~
+	  test -z "$concat_cmds" || concat_cmds=$concat_cmds!
 	  eval concat_cmds=\"\${concat_cmds}$reload_cmds $objlist $last_robj\"
 
 	  if ${skipped_export-false}; then
@@ -4086,7 +4086,7 @@
 	    $run $rm $export_symbols
 	    libobjs=$output
 	    # Append the command to create the export file.
-	    eval concat_cmds=\"\$concat_cmds~$export_symbols_cmds\"
+	    eval concat_cmds=\"\$concat_cmds!$export_symbols_cmds\"
           fi
 
 	  # Set up a command to remove the reloadale object files
@@ -4101,7 +4101,7 @@
 	  $echo "creating a temporary reloadable object file: $output"
 
 	  # Loop through the commands generated above and execute them.
-	  save_ifs="$IFS"; IFS='~'
+	  save_ifs="$IFS"; IFS='!'
 	  for cmd in $concat_cmds; do
 	    IFS="$save_ifs"
 	    $show "$cmd"
@@ -4128,9 +4128,9 @@
 
 	  # Append the command to remove the reloadable object files
 	  # to the just-reset $cmds.
-	  eval cmds=\"\$cmds~$rm $delfiles\"
+	  eval cmds=\"\$cmds!$rm $delfiles\"
 	fi
-	save_ifs="$IFS"; IFS='~'
+	save_ifs="$IFS"; IFS='!'
 	for cmd in $cmds; do
 	  IFS="$save_ifs"
 	  $show "$cmd"
@@ -4284,7 +4284,7 @@
 
       output="$obj"
       eval cmds=\"$reload_cmds\"
-      save_ifs="$IFS"; IFS='~'
+      save_ifs="$IFS"; IFS='!'
       for cmd in $cmds; do
 	IFS="$save_ifs"
 	$show "$cmd"
@@ -4320,7 +4320,7 @@
 	reload_objs="$libobjs $reload_conv_objs"
 	output="$libobj"
 	eval cmds=\"$reload_cmds\"
-	save_ifs="$IFS"; IFS='~'
+	save_ifs="$IFS"; IFS='!'
 	for cmd in $cmds; do
 	  IFS="$save_ifs"
 	  $show "$cmd"
@@ -5342,7 +5342,7 @@
 	      if test "$obj" = "$last_oldobj" ; then
 	        RANLIB=$save_RANLIB
 	      fi  
-	      test -z "$concat_cmds" || concat_cmds=$concat_cmds~
+	      test -z "$concat_cmds" || concat_cmds=$concat_cmds!
 	      eval concat_cmds=\"\${concat_cmds}$old_archive_cmds\"
 	      objlist=
 	    fi
@@ -5352,11 +5352,11 @@
 	  if test "X$oldobjs" = "X" ; then
 	    eval cmds=\"\$concat_cmds\"
 	  else
-	    eval cmds=\"\$concat_cmds~$old_archive_cmds\"
+	    eval cmds=\"\$concat_cmds!$old_archive_cmds\"
 	  fi
 	fi
       fi
-      save_ifs="$IFS"; IFS='~'
+      save_ifs="$IFS"; IFS='!'
       for cmd in $cmds; do
 	IFS="$save_ifs"
 	$show "$cmd"
@@ -5755,7 +5755,7 @@
 	  # Do each command in the postinstall commands.
 	  lib="$destdir/$realname"
 	  eval cmds=\"$postinstall_cmds\"
-	  save_ifs="$IFS"; IFS='~'
+	  save_ifs="$IFS"; IFS='!'
 	  for cmd in $cmds; do
 	    IFS="$save_ifs"
 	    $show "$cmd"
@@ -5978,7 +5978,7 @@
 
       # Do each command in the postinstall commands.
       eval cmds=\"$old_postinstall_cmds\"
-      save_ifs="$IFS"; IFS='~'
+      save_ifs="$IFS"; IFS='!'
       for cmd in $cmds; do
 	IFS="$save_ifs"
 	$show "$cmd"
@@ -6016,7 +6016,7 @@
 	if test -n "$finish_cmds"; then
 	  # Do each command in the finish commands.
 	  eval cmds=\"$finish_cmds\"
-	  save_ifs="$IFS"; IFS='~'
+	  save_ifs="$IFS"; IFS='!'
 	  for cmd in $cmds; do
 	    IFS="$save_ifs"
 	    $show "$cmd"
@@ -6293,7 +6293,7 @@
 	    if test -n "$library_names"; then
 	      # Do each command in the postuninstall commands.
 	      eval cmds=\"$postuninstall_cmds\"
-	      save_ifs="$IFS"; IFS='~'
+	      save_ifs="$IFS"; IFS='!'
 	      for cmd in $cmds; do
 		IFS="$save_ifs"
 		$show "$cmd"
@@ -6308,7 +6308,7 @@
 	    if test -n "$old_library"; then
 	      # Do each command in the old_postuninstall commands.
 	      eval cmds=\"$old_postuninstall_cmds\"
-	      save_ifs="$IFS"; IFS='~'
+	      save_ifs="$IFS"; IFS='!'
 	      for cmd in $cmds; do
 		IFS="$save_ifs"
 		$show "$cmd"
@@ -6778,8 +6778,8 @@
 
 # Commands used to build and install an old-style archive.
 RANLIB="ranlib"
-old_archive_cmds="\$AR \$AR_FLAGS \$oldlib\$oldobjs\$old_deplibs~\$RANLIB \$oldlib"
-old_postinstall_cmds="\$RANLIB \$oldlib~chmod 644 \$oldlib"
+old_archive_cmds="\$AR \$AR_FLAGS \$oldlib\$oldobjs\$old_deplibs!\$RANLIB \$oldlib"
+old_postinstall_cmds="\$RANLIB \$oldlib!chmod 644 \$oldlib"
 old_postuninstall_cmds=""
 
 # Create an old-style archive from a shared archive.
@@ -7074,8 +7074,8 @@
 
 # Commands used to build and install an old-style archive.
 RANLIB="ranlib"
-old_archive_cmds="\$AR \$AR_FLAGS \$oldlib\$oldobjs\$old_deplibs~\$RANLIB \$oldlib"
-old_postinstall_cmds="\$RANLIB \$oldlib~chmod 644 \$oldlib"
+old_archive_cmds="\$AR \$AR_FLAGS \$oldlib\$oldobjs\$old_deplibs!\$RANLIB \$oldlib"
+old_postinstall_cmds="\$RANLIB \$oldlib!chmod 644 \$oldlib"
 old_postuninstall_cmds=""
 
 # Create an old-style archive from a shared archive.
